### PR TITLE
Use graphics-core20 for graphics stack and stop using the obsolete snapcraft-desktop-helpers

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

**Review by Saviq — CHANGES_REQUESTED** [review]
I believe this needs fixing:

```plain
(xfce4-terminal:973840): GLib-GIO-WARNING **: 14:04:38.560: Error creating IO channel for /proc/self/mountinfo: Permission denied (g-file-error-quark, 2)                                               
                                                                                                                                                                                                        
(xfce4-terminal:973840): GLib-GIO-ERROR **: 14:04:41.710: No GSettings schemas are installed on the system                                                                                              
                                                                                                                                                                                                        
(epiphany:974062): GLib-GIO-ERROR **: 14:04:46.195: No GSettings schemas are installed on the system        
```

**Comment by AlanGriffiths** [comment]
> I believe this needs fixing:

I don't see those GLib-GIO-ERROR messages...
```plain
$ snap refresh --amend --channel edge/pr26 egmde-confined-desktop 
egmde-confined-desktop (edge/pr26) 143-mir2.4.1-snap1 from Alan Griffiths (alangriffiths) refreshed
$ egmde-confined-desktop 
[2021-08-27 14:58:03.050719] <information> mirserver: Starting
...
+ x-terminal-emulator
Failed to connect to session manager: Failed to connect to the session manager: Could not open network socket
Gdk-Message: 14:58:06.694: Unable to load hand2 from the cursor theme
error: socket path "/run/user/1000/snap.egmde-confined-desktop/webkitgtk-wayland-compositor-51ea1c20-98f4-4cca-a480-4d68b1c5545e" plus null terminator exceeds 108 bytes
Nested Wayland compositor could not create display socket

** (epiphany:239077): WARNING **: 14:58:16.331: Failed to search secrets in password schema: Could not connect: Permission denied

** (epiphany:239077): WARNING **: 14:58:16.434: Could not open /sys/class/dmi/id/chassis_type: Failed to open file “/sys/class/dmi/id/chassis_type”: Permission denied

** (epiphany:239077): WARNING **: 14:58:16.434: Could not open /sys/firmware/acpi/pm_profile: Failed to open file “/sys/firmware/acpi/pm_profile”: Permission denied

** (epiphany:239077): WARNING **: 14:58:16.621: Cannot fetch source for filter 1f353f7cdbb012b9fb1226455f1b3becba42070e1970c1524996fa3a871af406 from <https://easylist-downloads.adblockplus.org/easylist_min_content_blocker.json>: TLS/SSL support not available; install glib-networking

** (WebKitWebProcess:239110): WARNING **: 14:58:17.318: Could not open /sys/class/dmi/id/chassis_type: Failed to open file “/sys/class/dmi/id/chassis_type”: Permission denied

** (WebKitWebProcess:239110): WARNING **: 14:58:17.318: Could not open /sys/firmware/acpi/pm_profile: Failed to open file “/sys/firmware/acpi/pm_profile”: Permission denied
[2021-08-27 14:58:28.249372] <information> mirserver: Stopping
```

**Review by Saviq — CHANGES_REQUESTED** [review]
"Xfce Terminal" opens the settings dialog, not the terminal itself (as opposed to Ctrl+Alt+T).

"Web" results in:
```plain
** (epiphany:3199021): ERROR **: 12:44:16.861: Fatal initialization error: Failed to create directory “/var/snap/egmde-confined-desktop/5034/.local/usr/share/epiphany”.
```

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -152,24 +159,20 @@ parts:
...
+      - xfce4-terminal
+      - libgdk-pixbuf2.0-bin # GDK pixbuf
+      - libglib2.0-0
     override-build: |
       snapcraftctl build
-      # desktop-launch doesn't handle running without a user session very well, this avoids slow startup here
-      sed --in-place s'/^  needs_update=true/:/' $SNAPCRAFT_PART_INSTALL/bin/desktop-launch
+      update-mime-database $SNAPCRAFT_PART_INSTALL/usr/share/mime
+      rm -f $SNAPCRAFT_PART_INSTALL/usr/share/applications/exo-*
+      rm -f $SNAPCRAFT_PART_INSTALL/usr/share/applications/xfce4-terminal-settings.desktop
```
Comment (Saviq) [inline snap/snapcraft.yaml]: These are _prime_ examples for `prime: - -...`, care to make it so? Won't block. mind you.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -221,12 +213,47 @@ parts:
       - libpulse0
       - libaudio2
       - libsndio7.0
-      - va-driver-all
+
+  cleanup:
```
Comment (Saviq) [inline snap/snapcraft.yaml]: Should this still be named `cleanup` now that it does so much more?

**Review by Saviq — APPROVED** [review]
Works! Feel free to tweak what I wrote inline, or ignore it :)

bors d+

File: snap/snapcraft.yaml (hunk 2 above)
Comment (AlanGriffiths) [reply snap/snapcraft.yaml]: I don't care strongly enough to think of a new name.

**Comment by AlanGriffiths** [comment]
bors r=@Saviq

**Pull request merged**